src/components/Pagination/Pagination.jsx

```diff
@@ -1,163 +1,220 @@
-import React, { useState, useEffect } from 'react';
+import React, { useEffect, useReducer } from 'react';
 import PropTypes from 'prop-types';
 
-import { Flex, PseudoBox, useTheme } from '@chakra-ui/core';
+import { Flex, Button, useTheme } from '@chakra-ui/core';
 
 const LEFT_PAGE = 'LEFT';
 const RIGHT_PAGE = 'RIGHT';
+const SET_PAGINATION_CONFIG = 'SET_PAGINATION_CONFIG';
+const SET_CURRENT_PAGE = 'SET_CURRENT_PAGE';
+const SET_PAGE_NUMBERS = 'SET_PAGE_NUMBERS';
+
+/**
+ * @function range
+ *
+ * @param {number} from - Range start value
+ * @param {number} to - Range end value
+ * @param {number} step - Number to increment each loop count by
+ * @returns {Array.<number>} - An array of numbers
+ */
+
+const range = (from, to, step = 1) => {
+  let i = from;
+  const range = [];
+
+  for (let i = from; i <= to; i = i + step) {
+    range.push(i);
+  }
+
+  return range;
+};
 
 /**
  * @function Pagination
  *
  * @param {number} totalRecords - Total records count
- * @param {function} onPageChanged - Callback function that passes paginated information
  * @param {number} pageLimit - Number of items to display per page
  * @param {number} pageNeighbors - Number of neighboring pages when pagination is in the middle of the list
+ * @param {function} onPageChanged - Callback function that passes paginated information
  *
  */
 
+function reducer(state, action) {
+  switch (action.type) {
+    case SET_PAGINATION_CONFIG:
+      return {
+        ...state,
+        ...action.payload,
+      };
```

Comment:
The idea here isn't to have a catch-all. Rather, the syntax is:

```js
case EVENT_NAME:
  // handle specific event
```

So say you were updating the page limit based on the dependencies changing, might look like:

```js
case UPDATE_PAGE_LIMIT:
  // handle only stuff for updating the page limit
```

> Also, you're not restricted to using just `action.payload` just fyi.

Reply:
In this case, should I have 5 separate dispatches for each event for the useEffect on line 78